File: frontend/src/api/contacts.ts
import api from './client'

export interface ContactItem {
  id: number
  username: string
  email: string
  full_name: string | null
  department_name: string | null
  phone: string | null
}

export interface DepartmentTreeNode {
  id: number
  name: string
  children: DepartmentTreeNode[]
  employee_count: number
}

export interface PaginatedContacts {
  items: ContactItem[]
  total: number
  page: number
  page_size: number
}

export const contactsApi = {
  list: (params?: { search?: string; department_id?: number; page?: number; page_size?: number }) =>
    api.get<PaginatedContacts>('/contacts', { params }),

  tree: () =>
    api.get<DepartmentTreeNode[]>('/contacts/tree'),
}

File: backend/app/schemas/contacts.py
from pydantic import BaseModel, ConfigDict


class ContactOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    username: str
    email: str
    full_name: str | None = None
    department_name: str | None = None
    phone: str | None = None


class DepartmentTreeNode(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    children: list["DepartmentTreeNode"] = []
    employee_count: int = 0


class PaginatedContacts(BaseModel):
    items: list[ContactOut]
    total: int
    page: int
    page_size: int
